### ISSUE TITLE
On the performance of SWformer-VO1 as well as SWformer-VO2

### ISSUE THREAD

**Issue by Nemophilist98** [ZhuYaoHui1998/SWformer-VO#3]
Dear Author: This is the performance of SWformer-VO1 on the test set, but the performance doesn't seem to reach the results in the paper, especially the first dataset, with my parameter settings placed behind it, can you please give a little advice?

<img width="402" alt="1" src="https://github.com/user-attachments/assets/06c90296-b47a-4a18-a217-aa9c3bc22e54">
<img width="473" alt="3" src="https://github.com/user-attachments/assets/03ec4efe-752f-4ac8-a6e7-60b6225381e1">
<img width="128" alt="4" src="https://github.com/user-attachments/assets/3dd28608-2e80-45f7-b6fa-a63c0679d68e">
<img width="324" alt="5" src="https://github.com/user-attachments/assets/d42e09f6-eb6c-46c3-8fdd-7dd79b184e5f">
<img width="193" alt="6" src="https://github.com/user-attachments/assets/c6127063-0fec-4e7f-92e8-f5604dfccec0">
<img width="631" alt="7" src="https://github.com/user-attachments/assets/e81726ae-081c-474f-8196-5ad01bf62077">

```
    args = {
        "data_dir": "data",
        "bsize": 8,  # batch size
        "val_split": 0.1,  # percentage to use as validation data
        "window_size": 2,  # number of frames in window
        "overlap": 1,  # number of frames overlapped between windows
        "optimizer": "Adam",  # optimizer [Adam, SGD, Adagrad, RAdam]
        "lr": 1e-5,  # learning rate
        "momentum": 0.9,  # SGD momentum
        "weight_decay": 1e-4,  # SGD momentum
        "epoch": 100,  # train iters each timestep
    	"weighted_loss": None,  # float to weight angles in loss function
      	"pretrained_ViT": False,  # load weights from pre-trained ViT
        "checkpoint_path": "checkpoints/1128",  # path to save checkpoint
        "checkpoint":"checkpoint_e70.pth",  # checkpoint
    }

    # tiny  - patch_size=16, embed_dim=192, depth=12, num_heads=3
    # small - patch_size=16, embed_dim=384, depth=12, num_heads=6
    # base  - patch_size=16, embed_dim=768, depth=12, num_heads=12
    model_params = {
        "dim": 192,
        "image_size": (224, 678),  #(192, 640),  448 678
        "patch_size": 7,
        "attention_type": 'divided_space_time',  # ['divided_space_time', 'space_only','joint_space_time', 'time_only']
        "num_frames": args["window_size"],
        "sw_windows_size":8,
        "num_classes": 6 * (args["window_size"] - 1),  # 6 DoF for each frame
        "depth": 28,
        "heads": 28,
        "sw_depth": [12,8,8],
        "sw_num_heads":[12,12,12],
        "dim_head": 64,
        "attn_dropout": 0.1,
        "ff_dropout": 0.1,
        "time_only": False,
    }
```

Also when training SWformer-VO2, the SWformer-VO2 model failed to converge during my training. The parameters I set are as follows (referring to the paper):

```
args = {
        "data_dir": "data",
        "bsize": 8,  # batch size
        "val_split": 0.1,  # percentage to use as validation data
        "window_size": 2,  # number of frames in window
        "overlap": 1,  # number of frames overlapped between windows
        "optimizer": "Adam",  # optimizer [Adam, SGD, Adagrad, RAdam]
        "lr": 1e-5,  # learning rate
        "momentum": 0.9,  # SGD momentum
        "weight_decay": 1e-4,  # SGD momentum
        "epoch": 100,  # train iters each timestep
    	"weighted_loss": None,  # float to weight angles in loss function
      	"pretrained_ViT": False,  # load weights from pre-trained ViT
        "checkpoint_path": "checkpoints/1128",  # path to save checkpoint
        "checkpoint":None,  # checkpoint
    }

    # tiny  - patch_size=16, embed_dim=192, depth=12, num_heads=3
    # small - patch_size=16, embed_dim=384, depth=12, num_heads=6
    # base  - patch_size=16, embed_dim=768, depth=12, num_heads=12
    model_params = {
        "dim": 192,
        "image_size": (448, 678),  #(192, 640),  448 678 224 678
        "patch_size": 7,
        "attention_type": 'divided_space_time',  # ['divided_space_time', 'space_only','joint_space_time', 'time_only']
        "num_frames": args["window_size"],
        "sw_windows_size":8,
        "num_classes": 6 * (args["window_size"] - 1),  # 6 DoF for each frame
        "depth": 28,
        "heads": 28,
        "sw_depth": [12,8,8],
        "sw_num_heads":[12,12,12],
        "dim_head": 64,
        "attn_dropout": 0.1,
        "ff_dropout": 0.1,
        "time_only": False,
    }
```

**Comment by ZhuYaoHui1998** [comment]
bsize set to 2 . could you try this












Marlboro
***@***.***






---Original---
From: ***@***.***&gt;
Date: Sat, Nov 30, 2024 14:13 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [ZhuYaoHui1998/SWformer-VO] On the performance of SWformer-VO1 aswell as SWformer-VO2 (Issue #3)




 
Dear Author: This is the performance of SWformer-VO1 on the test set, but the performance doesn't seem to reach the results in the paper, especially the first dataset, with my parameter settings placed behind it, can you please give a little advice?
 1.png (view on web) 3.png (view on web) 4.png (view on web) 5.png (view on web) 6.png (view on web) 7.png (view on web)     args = {         "data_dir": "data",         "bsize": 8,  # batch size         "val_split": 0.1,  # percentage to use as validation data         "window_size": 2,  # number of frames in window         "overlap": 1,  # number of frames overlapped between windows         "optimizer": "Adam",  # optimizer [Adam, SGD, Adagrad, RAdam]         "lr": 1e-5,  # learning rate         "momentum": 0.9,  # SGD momentum         "weight_decay": 1e-4,  # SGD momentum         "epoch": 100,  # train iters each timestep     	"weighted_loss": None,  # float to weight angles in loss function       	"pretrained_ViT": False,  # load weights from pre-trained ViT         "checkpoint_path": "checkpoints/1128",  # path to save checkpoint         "checkpoint":"checkpoint_e70.pth",  # checkpoint     }     # tiny  - patch_size=16, embed_dim=192, depth=12, num_heads=3     # small - patch_size=16, embed_dim=384, depth=12, num_heads=6     # base  - patch_size=16, embed_dim=768, depth=12, num_heads=12     model_params = {         "dim": 192,         "image_size": (224, 678),  #(192, 640),  448 678         "patch_size": 7,         "attention_type": 'divided_space_time',  # ['divided_space_time', 'space_only','joint_space_time', 'time_only']         "num_frames": args["window_size"],         "sw_windows_size":8,         "num_classes": 6 * (args["window_size"] - 1),  # 6 DoF for each frame         "depth": 28,         "heads": 28,         "sw_depth": [12,8,8],         "sw_num_heads":[12,12,12],         "dim_head": 64,         "attn_dropout": 0.1,         "ff_dropout": 0.1,         "time_only": False,     }  
Also when training SWformer-VO2, the SWformer-VO2 model failed to converge during my training. The parameters I set are as follows (referring to the paper):
 args = {         "data_dir": "data",         "bsize": 8,  # batch size         "val_split": 0.1,  # percentage to use as validation data         "window_size": 2,  # number of frames in window         "overlap": 1,  # number of frames overlapped between windows         "optimizer": "Adam",  # optimizer [Adam, SGD, Adagrad, RAdam]         "lr": 1e-5,  # learning rate         "momentum": 0.9,  # SGD momentum         "weight_decay": 1e-4,  # SGD momentum         "epoch": 100,  # train iters each timestep     	"weighted_loss": None,  # float to weight angles in loss function       	"pretrained_ViT": False,  # load weights from pre-trained ViT         "checkpoint_path": "checkpoints/1128",  # path to save checkpoint         "checkpoint":None,  # checkpoint     }     # tiny  - patch_size=16, embed_dim=192, depth=12, num_heads=3     # small - patch_size=16, embed_dim=384, depth=12, num_heads=6     # base  - patch_size=16, embed_dim=768, depth=12, num_heads=12     model_params = {         "dim": 192,         "image_size": (448, 678),  #(192, 640),  448 678 224 678         "patch_size": 7,         "attention_type": 'divided_space_time',  # ['divided_space_time', 'space_only','joint_space_time', 'time_only']         "num_frames": args["window_size"],         "sw_windows_size":8,         "num_classes": 6 * (args["window_size"] - 1),  # 6 DoF for each frame         "depth": 28,         "heads": 28,         "sw_depth": [12,8,8],         "sw_num_heads":[12,12,12],         "dim_head": 64,         "attn_dropout": 0.1,         "ff_dropout": 0.1,         "time_only": False,     }
